src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -80,17 +83,23 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
 
             var semanticVersion = await GetDocumentSemanticVersionAsync(documentSnapshot).ConfigureAwait(false);
 
-            // If we have a matching cached response, avoid computation and return early.
-            if (_cachedResponses.TryGetValue(textDocumentIdentifier.Uri, out var response) &&
-                response.IsCSharpFinalized &&
-                response.SemanticVersion == semanticVersion &&
-                response.Range == range)
+            // See if we can use our cache to at least partially avoid recomputation.
+            var cachedTokens = await GetCachedTokensAsync(
+                textDocumentIdentifier.Uri, documentSnapshot, documentVersion, semanticVersion, range, _logger, cancellationToken).ConfigureAwait(false);
```

Comment:
Instead of using the URI as a cache lookup can we use the document snapshots filepath?

Reply:
Is there a reason the file path would be preferable to the URI as a cache lookup? Is it because we'd be passing in less parameters?

Reply:
> Is there a reason the file path would be preferable to the URI as a cache lookup? Is it because we'd be passing in less parameters?

I guess that's one reason. I don't feel strongly. The root concern for my comment was from the interesting dance we have to do in order to tie the LSP and non-LSP worlds together (uri vs. document snapshot). There isn't an easy way to do that so I was just spitballing. If this works it's less work to leave it as its :)